homeassistant/components/cover/tuya.py

```diff
@@ -0,0 +1,60 @@
+"""
+Support for Tuya cover.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/components/cover.tuya/
+"""
+from homeassistant.components.cover import (
+    CoverDevice, ENTITY_ID_FORMAT, SUPPORT_OPEN, SUPPORT_CLOSE, SUPPORT_STOP)
+from homeassistant.components.tuya import DATA_TUYA, TuyaDevice
+
+DEPENDENCIES = ['tuya']
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up Tuya cover devices."""
+    if discovery_info is None:
+        return
+    tuya = hass.data[DATA_TUYA]
+    dev_ids = discovery_info.get('dev_ids')
+    devices = []
+    for dev_id in dev_ids:
+        device = tuya.get_device_by_id(dev_id)
+        if device is None:
+            continue
+        devices.append(TuyaCover(device))
+    add_devices(devices)
+
+
+class TuyaCover(TuyaDevice, CoverDevice):
+    """Tuya cover devices."""
+
+    def __init__(self, tuya):
+        """Init tuya cover device."""
+        super().__init__(tuya)
+        self.entity_id = ENTITY_ID_FORMAT.format(tuya.object_id())
+
+    @property
+    def supported_features(self):
+        """Flag supported features."""
+        supported_features = SUPPORT_OPEN | SUPPORT_CLOSE
+        if self.tuya.support_stop():
+            supported_features |= SUPPORT_STOP
+        return supported_features
+
+    @property
+    def is_closed(self):
+        """Return if the cover is closed or not."""
+        return None
```

Comment:
I'm not sure this is the best way. I assume tuya covers can't report if they are open or closed. Then the question is: How is that best handled in home assistant?

If you have an example of a previous implementation, please link to that.

Ill dig around a bit myself.

Reply:
Ok, sounds good.